slang-gfx.h

```diff
@@ -418,14 +429,11 @@ struct ResourceStateSet
 
 
 /// Describes how memory for the resource should be allocated for CPU access.
-struct MemoryType
+enum class MemoryType
 {
-    enum Enum
-    {
-        GpuOnly = 0x0,
-        CpuRead = 0x1,
-        CpuWrite = 0x2
-    };
+    DeviceLocal,
+    Upload,
+    ReadBack,
```

Comment:
No, that would complicate things. A resource can only reside in one type of heap at a time. It is cleaner for gfx to not over-manage things by automatically creating more than one underlying resources.

Reply:
I'm not advocating for something to have multiple allocations. I'm advocating that if the user wants something that resides in memory that is both CPU-readable *and* CPU-writable, I would have assumed we can give them a single allocation that supports both.
Am I just out of date in my understanding of GPUs? Is there really no guarantee that there is a space that can be used for both upload and readback? I guess that isn't commonly needed, so maybe it is okay to avoid.

Reply:
I see what you mean. It is possible to read a Upload heap from cpu, just slow. So the access flag doesn't truly tell which heap to create resource from. There is also the custom heap, which isn't exposed by gfx and isn't guaranteed supported by the driver, that you can create resources that are both readable, writable and gpu accessible.